includes/admin/load.php

```diff
@@ -204,51 +210,75 @@ function perflab_dismiss_wp_pointer_wrapper() {
  * Callback function to handle admin scripts.
  *
  * @since 2.8.0
+ * @since n.e.x.t Renamed to perflab_enqueue_features_page_scripts().
  */
-function perflab_enqueue_modules_page_scripts() {
-	wp_enqueue_script( 'updates' );
-
-	wp_localize_script(
-		'updates',
-		'_wpUpdatesItemCounts',
-		array(
-			'settings' => array(
-				'totals' => wp_get_update_data(),
-			),
-		)
-	);
-
+function perflab_enqueue_features_page_scripts() {
 	wp_enqueue_script( 'thickbox' );
 	wp_enqueue_style( 'thickbox' );
+}
 
-	wp_enqueue_script( 'plugin-install' );
-
-	wp_enqueue_script(
-		'perflab-plugin-management',
-		plugin_dir_url( __FILE__ ) . 'js/perflab-plugin-management.js',
-		array( 'jquery' ),
-		'1.0.0',
-		array(
-			'in_footer' => true,
-			'strategy'  => 'defer',
-		)
-	);
+/**
+ * Register admin actions for handling performance feature plugin installation/activation.
+ *
+ * @since n.e.x.t
+ */
+function perflab_register_admin_actions() {
+	add_action( 'admin_action_perflab_install_activate_plugins', 'perflab_install_activate_plugins_callback' );
 }
+add_action( 'admin_init', 'perflab_register_admin_actions' );
 
 /**
- * Callback function hooked to admin_action_perflab_activate_plugin to handle plugin activation.
+ * Callback for handling installation/activation of plugin.
  *
- * @since 2.8.0
+ * @since n.e.x.t
  */
-function perflab_activate_plugin() {
-	// Do not proceed if plugin query arg is not present.
-	if ( empty( $_GET['plugin'] ) ) {
-		return;
+function perflab_install_activate_plugins_callback() {
+	check_admin_referer( 'perflab_install_activate_plugins' );
+
+	require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-ajax-upgrader-skin.php';
+
+	if ( ! isset( $_GET['slug'] ) ) {
+		wp_die( esc_html__( 'Missing required parameter.', 'performance-lab' ) );
+	}
+
+	if ( ! current_user_can( 'install_plugins' ) || ! current_user_can( 'activate_plugins' ) ) {
+		wp_die( esc_html__( 'Sorry, you are not allowed to manage plugins for this site. Please contact the administrator.', 'performance-lab' ) );
+	}
```

Comment:
Related to my previous feedback: We should check permissions more granularly. A few things here:
* Please use `activate_plugin` (singular) with the plugin basename instead of the generic `activate_plugins`.
* We should only check for the `install_plugins` capability if the plugin isn't already installed yet.

Based on my other recommendation to pass the `file` (which contains the basename _if_ the plugin is already installed), we can simplify this function quite a bit:
1. If `file` is empty, we know the plugin needs to be installed. Otherwise we know the plugin only needs to be activated.
2. If the plugin needs to be installed, we need to check the `install_plugins` capability. Only then we run all the installation specific logic.
3. Then, we determine the plugin basename (only relevant if `file` is empty, i.e. after installation of the plugin.
4. Then we check for the `activate_plugin` capability passing the basename. Then we can run the activation logic.